### PR TITLE
Allow skipping webOnlyInitializePlatform in Flutter for Web

### LINKED ISSUE #40389: [web] Release build is not registering web plugins
For some reason `hasPlugins`  is `false` on release builds. So the wrong entrypoint gets built.

https://github.com/flutter/flutter/blob/963c8a89213ad13a472b799d6d72e03cbe629cf7/packages/flutter_tools/lib/src/build_runner/build_script.dart#L375

```
[✓] Flutter (Channel master, v1.10.2-pre.59, on Mac OS X 10.14.6 18G95, locale en-CH)

[✓] Android toolchain - develop for Android devices (Android SDK version 28.0.3)
[✓] Xcode - develop for iOS and macOS (Xcode 10.3)
[✓] Chrome - develop for the web
[✓] Android Studio (version 3.5)
[✓] VS Code (version 1.31.1)
[✓] Connected device (3 available)

• No issues found!
```

### PR DESCRIPTION
**Pull request by jonahwilliams**
## Description

Allow configuring whether `webOnlyInitializePlatform` is invoked by the shell automatically with the flag (--no)--web-initialize-platform to flutter run and build. Also fixes an issue with flutter build not checking for plugins.

cc @jacob314 PTAL and let me know if this is sufficient for your usecase

Also fixes https://github.com/flutter/flutter/issues/40389

### CODE REVIEW

**Comment by jacob314** [comment]
Works like a charm! This will make my job incrementally porting from dart:html much easier!

File: packages/flutter_tools/lib/src/build_runner/build_script.dart [hunk 1]
```diff
@@ -383,7 +390,9 @@ import "${path.url.basename(buildStep.inputId.path)}" as entrypoint;
 
 Future<void> main() async {
   registerPlugins(webPluginRegistry);
```
Comment (jacob314) [inline packages/flutter_tools/lib/src/build_runner/build_script.dart]: doesn't matter for my use case but should this initialization line also be inside the if block?

File: packages/flutter_tools/lib/src/build_runner/build_script.dart [hunk 2]
```diff
@@ -394,7 +403,9 @@ import 'dart:ui' as ui;
 import "${path.url.basename(buildStep.inputId.path)}" as entrypoint;
 
 Future<void> main() async {
-  await ui.webOnlyInitializePlatform();
+  if ($initializePlatform) {
```
Comment (jacob314) [inline packages/flutter_tools/lib/src/build_runner/build_script.dart]: doesn't really matter as it will be tree shaken anyway but you could write
${initializePlatform ? 'await ui.webOnlyInitializePlatform();' : ''}
to generate code with the initialize line omitted depending on the style you are going for with this generated code.

File: packages/flutter_tools/lib/src/build_runner/web_compilation_delegate.dart [hunk 3]
```diff
@@ -28,18 +31,23 @@ class BuildRunnerWebCompilationProxy extends WebCompilationProxy {
...
-    String projectName
+    String projectName,
+    bool initializePlatform,
   }) async {
     // Create the .dart_tool directory if it doesn't exist.
     projectDirectory
       .childDirectory('.dart_tool')
       .createSync();
+    final FlutterProject flutterProject = FlutterProject.fromDirectory(projectDirectory);
+    final bool hasWebPlugins = findPlugins(flutterProject)
```
Comment (jacob314) [inline packages/flutter_tools/lib/src/build_runner/web_compilation_delegate.dart]: is this hasWebPlugins line an unrelated bug fix?

**Review by jacob314 — COMMENTED** [review]
lgtm once there is a test.

**Review by jacob314 — COMMENTED** [review]
lgtm
with a couple small nits. Thanks for implementing this so quickly!

**Review by harryterkelsen — APPROVED** [review]
